modules/storages/spec/common/peripherals/storage_requests_spec.rb

```diff
@@ -310,4 +310,174 @@ def origin_name
+      before do
+        allow(connection_manager).to receive(:get_access_token).and_return(ServiceResult.failure)
+      end
+
+      it 'must return ":not_authorized" ServiceResult' do
+        expect(subject.download_link_query(user:)).to be_failure
+      end
+    end
+
+    describe 'with first outbound request returning 200 and an empty body' do
+      before do
+        stub_request(:post, "#{url}/ocs/v2.php/apps/files_sharing/api/v1/shares")
+          .with(
+            body: hash_including(
+              {
+                shareType: 3,
+                path: query_payload.parent,
+                expireDate: Date.tomorrow.iso8601
+              }
+            )
+          )
+          .to_return(status: 200)
+      end
+
+      it 'must return :not_authorized ServiceResult' do
+        subject
+          .upload_link_query(user:, finalize_url: nil)
+          .match(
+            on_success: ->(query) do
+              result = query.call(query_payload)
+              expect(result).to be_failure
+              expect(result.errors.code).to be(:not_authorized)
+            end,
+            on_failure: ->(error) do
+              raise "Files query could not be created: #{error}"
+            end
+          )
+      end
+    end
+
+    describe 'with second outbound request returning 200 and an empty body' do
+      before do
+        stub_request(:put, "#{url}/ocs/v2.php/apps/files_sharing/api/v1/shares/#{share_id}")
+          .with(body: { permissions: 4 })
+          .to_return(status: 200)
+      end
+
+      it 'must return :not_authorized ServiceResult' do
+        subject
+          .upload_link_query(user:, finalize_url: nil)
+          .match(
+            on_success: ->(query) do
+              result = query.call(query_payload)
+              expect(result).to be_failure
+              expect(result.errors.code).to be(:not_authorized)
+            end,
+            on_failure: ->(error) do
+              raise "Files query could not be created: #{error}"
+            end
+          )
+      end
+    end
+
+    shared_examples_for 'outbound is failing' do |method = :get, path = '', code = 500, symbol = :error|
+      describe "with outbound request returning #{code}" do
+        before do
+          stub_request(method, "#{url}/ocs/v2.php/apps/files_sharing/api/v1/shares#{path}").to_return(status: code)
+        end
+
+        it "must return :#{symbol} ServiceResult" do
+          subject
+            .upload_link_query(user:, finalize_url: nil)
+            .match(
+              on_success: ->(query) do
+                result = query.call(query_payload)
+                expect(result).to be_failure
+                expect(result.errors.code).to be(symbol)
+              end,
+              on_failure: ->(error) do
+                raise "Files query could not be created: #{error}"
+              end
+            )
+        end
+      end
+    end
+
+    it_behaves_like 'outbound is failing', :post, '', 404, :not_found
```

Comment:
You can have a less cluttered output by using `include_examples` instead of `it_behaves_like`, especially when running tests with `rspec --format documentation`.

* with `include_examples`, test description is `Storages::Peripherals::StorageRequests#upload_link_query with outbound request returning 404 must return :not_found ServiceResult`
* with `it_behaves_like`, test description is `Storages::Peripherals::StorageRequests#upload_link_query behaves like outbound is failing with outbound request returning 404 must return :not_found ServiceResult`

Reply:
Fixed.